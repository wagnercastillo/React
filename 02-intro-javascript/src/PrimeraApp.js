import React from "react";
import PropTypes from "prop-types";

const PrimeraApp = ({ saludo, subtitulo }) => {
 
  return (
    <>
      <h1>{saludo}</h1>
      <p>{subtitulo}</p>
    </>
  );
};

PrimeraApp.defaultProps = {
  subtitulo: "Mi subtitulo",
};

PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired,
};

export default PrimeraApp;
