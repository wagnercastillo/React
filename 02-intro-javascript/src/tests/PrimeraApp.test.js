import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import PrimeraApp from "../PrimeraApp";

describe("Pruebas en <PrimeraApp/ >", () => {
	// test('Debe de mostrar el mensaje "Hola, soy goku" ', () => {
	// 	const saludo = "Hola, soy goku";
	// 	const { getByText } = render(<PrimeraApp saludo={saludo} />);
	// 	expect(getByText(saludo)).toBeInTheDocument();
	// });

	test("Debe de mostrar <PrimeraApp /> correctamente ", () => {
		const saludo = "Hola, soy goku";
		const wrapper = shallow(<PrimeraApp saludo={saludo} />);

		expect(wrapper).toMatchSnapshot();
	});

	test("Debe de mostrar el subtitulo enviado por props", () => {
		const saludo = "Hola, soy goku";
		const subtitulo = "Soy un subtitulo";
		const wrapper = shallow(
			
            <PrimeraApp saludo={saludo} subtitulo={subtitulo}
            
            />
		);
        
        const textoParrafo = wrapper.find("p").text();
        expect(textoParrafo).toBe(subtitulo);
	});
});
