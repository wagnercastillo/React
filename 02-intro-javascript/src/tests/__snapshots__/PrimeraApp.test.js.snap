// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/ > Debe de mostrar <PrimeraApp /> correctamente  1`] = `
<Fragment>
  <h1>
    Hola, soy goku
  </h1>
  <p>
    Mi subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas en <PrimeraApp/ > Debe de mostrar el subtitulo enviado por props 1`] = `
<Fragment>
  <h1>
    Hola, soy goku
  </h1>
  <p>
    Mi subtitulo
  </p>
</Fragment>
`;
